sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java

```diff
@@ -20,25 +24,49 @@
  */
 public class RetryPolicy implements HttpPipelinePolicy {
 
-    private static final String RETRY_AFTER_MS_HEADER = "retry-after-ms";
-
     private final ClientLogger logger = new ClientLogger(RetryPolicy.class);
     private final RetryStrategy retryStrategy;
 
+    private final HttpHeaderName retryAfterHeader;
+
     /**
      * Creates a default {@link ExponentialBackoff} retry policy.
      */
     public RetryPolicy() {
-        this(new ExponentialBackoff());
+        this(new ExponentialBackoff(), HttpHeaderName.AZURE_RETRY_AFTER_MS_HEADER);
+    }
+
+    /**
+     * Creates a default {@link ExponentialBackoff} retry policy.
+     * @param retryAfterHeader The retry after http header name to be used get retry after value from
+     * {@link HttpResponse}.
+     * @throws NullPointerException if {@code retryAfterHeader} is {@code null}.
+     */
+    public RetryPolicy(HttpHeaderName retryAfterHeader) {
```

Comment:
Why did you choose an enum rather than a string?

Reply:
I started with String, I already have two string to define, We will have more to use. Enum allow to ensure valid string always and not worried about passing invalid String values, thus defined as type .  We will need more values later https://github.com/google/guava/blob/master/guava/src/com/google/common/net/HttpHeaders.java

Reply:
That link is to a series of strings, which would be a better approach than an enum.

Reply:
Okay I will go with String.